src/library/components/stem-finder-result.js

```diff
@@ -134,10 +134,27 @@ var StemFinderResult = Component({
     var resource = this.props.resource
     var options = {href: resource.stem_resource_url}
 
+    // truncate title and/or description if they are too long for resource card height
+    var resourceName = resource.name;
+    var maxCharName = 125
+    if (resourceName.length >= maxCharName) {
+      resourceName = resourceName.substring(0, resourceName.lastIndexOf(' ', maxCharName)) + '...'
+      resourceName = resourceName.replace(/[^\w\s]\.\.\./, '...') // trim extraneous punctuation before ellipsis
+    }
+    var shortDesc = resource.filteredShortDescription
+    var maxCharDesc = 320
+    if (shortDesc.length + resource.name.length >= maxCharDesc) { // use full resource name for 'back' of card
+      shortDesc = shortDesc.substring(0, shortDesc.lastIndexOf(' ', maxCharDesc - resource.name.length)) + '...'
+      shortDesc = shortDesc.replace(/[^\w\s]\.\.\./, '...') // trim extraneous punctuation before ellipsis
+    }
+
```

Comment:
Not sure if it's applicable in this case and you might know about it, but quite often is also possible to do it using just CSS, e.g. using rules like this:
```css
    white-space: nowrap; 
    overflow: hidden;
    text-overflow: ellipsis;
```
and perhaps some width / max-width definiton. I've learned about it recently.

Reply:
@emcelroy would @pjanik's css approach work here? It looks like you are trying to be a bit smarter by breaking on an space character is that necessary?  If someone uses a long word in their title or description near the end this could cause an unexpected effect. So perhaps you want to include both approaches. Then most of the time it will break on a space, but in a few cases it will break at the overflow point.  

Also you probably want to use the ellipsis character not just 3 dots. 

If you do need the space breaking approach, then I'd recommend moving this code into a function. We have a similar function here already: https://github.com/concord-consortium/portal-pages/blob/9e5aec0a4eb18fcbd4b1e1ec79d6baa313b711fa/src/library/helpers/portal-object-helpers.js#L3-L14
You can extend that with an optional parameter to break on space, or just make a new function.

Reply:
@pjanik, @scytacki: Understood about converting to JSX and ES6 style. I will do that going forward. Please bear with me as I get up to speed on it.

Thanks for suggesting the CSS option. Yes, I experimented with that option but found it didn't work very well for multi-line text. It works best for text on a single line.

There are some semi-convoluted ways to make it work for multi-line text, but in the end, the JavaScript option made the most sense to me. I'd like to stick with that unless you feel strongly that there's a better option?

Good point on using the ellipsis character. I'll make that change.

Breaking on a space character makes things neater in terms of legibility IMO. I find it irritating to read words that bre... I'd much prefer to read words that... Again, if you strongly disagree we can change that, though maybe we should get a larger sampling of opinions first?

I think this code should prevent the case you describe with a long word because it essentially trims the text to the desired length then "moves back" to the nearest space character. That's what is intended anyway.

I can move that into a function or augment the existing one you pointed out. Thanks.

Reply:
You are right about the long word case. I didn't understand what the 'fromIndex' argument to `lastIndexOf` actually did.  I think it would be good to move this into a function. It will hopefully make it easier for others to figure out what it is doing.